fvm/scriptEnv.go

```diff
@@ -685,3 +686,35 @@ func (e *ScriptEnv) Commit() ([]programs.ContractUpdateKey, error) {
 func (e *ScriptEnv) AllocateStorageIndex(_ []byte) (uint64, error) {
 	return 0, errors.NewOperationNotSupportedError("AllocateStorageIndex")
 }
+
+func (e *ScriptEnv) BLSVerifyPOP(pk *runtime.PublicKey, sig []byte) (bool, error) {
+	key, err := crypto.DecodePublicKey(crypto.BLSBLS12381, pk.PublicKey)
```

Comment:
Not sure how the upper layers of Cadence would handle returned errors. But I know we would like to differentiate 2 kind of errors:
 - recoverable errors: the crypto package returns a sentinel error if any input is invalid (caller mistake) and the node shouldn't crash.
 - non recoverable errors: an unexpected exception during execution, that can cause the node to crash.
 
 This is done here for instance https://github.com/onflow/flow-go/blob/cadence-issue-1127/fvm/crypto/crypto.go#L104-L109.
 
 cc @janezpodhostnik

Reply:
> recoverable errors: the crypto package returns a sentinel error if any input is invalid (caller mistake) and the node shouldn't crash.

Would you want these functions to return `nil` on an invalid input?